manager/integration/tests/test_metric.py

```diff
@@ -0,0 +1,159 @@
+import pytest
+import requests
+from prometheus_client.parser import text_string_to_metric_families
+from common import client, core_api, volume_name  # NOQA
+from common import get_self_host_id
+from common import create_pv_for_volume, create_pvc_for_volume
+from common import create_and_check_volume, wait_for_volume_healthy
+from common import write_volume_random_data
+from common import wait_for_volume_detached, wait_for_volume_degraded
+from common import wait_for_volume_faulted, crash_replica_processes
+from common import Mi
+from common import wait_for_volume_detached_unknown
+from common import LONGHORN_NAMESPACE
+
+longhorn_volume_state = {
+    "creating": 1.0,
+    "attached": 2.0,
+    "detached": 3.0,
+    "attaching": 4.0,
+    "detaching": 5.0,
+    "deleting": 6.0,
+    }
+
+longhorn_volume_robustness = {
+    "unknown": 0.0,
+    "healthy": 1.0,
+    "degraded": 2.0,
+    "faulted": 3.0,
+}
+
+
+def get_metrics(core_api): # NOQA
+    lht_hostId = get_self_host_id()
+
+    pods = core_api.list_namespaced_pod(namespace="longhorn-system",
+                                        label_selector="app=longhorn-manager")
+    for pod in pods.items:
+        if pod.spec.node_name == lht_hostId:
+            manager_ip = pod.status.pod_ip
+            break
+
+    metrics = requests.get("http://{}:9500/metrics".format(manager_ip)).content
+    string_data = metrics.decode('utf-8')
+    result = text_string_to_metric_families(string_data)
+    return result
+
+
+def check_volume_metric(core_api, metric_name, metric_labels, expect_value=None): # NOQA
+    metric_data = get_metrics(core_api)
+    found = False
+    for family in metric_data:
+        for sample in family.samples:
+            if sample.name == metric_name:
+                found = True
+                item = sample
+                break
+
+    assert found is True
```

Comment:
Assert `item` is not `None` might achieve the same thing.

Reply:
Updated